File: 0209_Minimum Size Subarray Sum/MinimumSizeSubarraySum.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        var sum = 0;
        var l = 0;
        var ans = nums.Length + 1;
        for(int r = 0;r<nums.Length;r++)
        {
            sum += nums[r];
            
            while(sum >= target)
            {
                ans = Math.Min(r - l + 1, ans);
                sum -= nums[l++];
            }
        }
        
        return ans == nums.Length + 1 ? 0 : ans;
    }
}

File: 0209_Minimum Size Subarray Sum/MinimumSizeSubarraySum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int ans = Integer.MAX_VALUE;
        int l = 0, sum = 0;
        for(int r = 0;r<nums.length;r++) {
            sum += nums[r];
            while(sum >= target) {
                ans = Math.min(ans, r - l + 1);
                sum -= nums[l++];
            }
        }

        return ans == Integer.MAX_VALUE ? 0 : ans;
    }
}